packages/nx/src/tasks-runner/life-cycle.ts

```diff
@@ -124,4 +125,14 @@ export class CompositeLifeCycle implements LifeCycle {
       }
     }
   }
+
+  getTaskResults(): Record<string, TaskResult> {
+    let results = {};
+    for (let l of this.lifeCycles) {
+      if (l.getTaskResults) {
+        results = { ...results, ...l.getTaskResults() };
+      }
+    }
+    return results;
+  }
```

Comment:
I don't think we need to add this here. It doesn't need to be on every life cycle. Just one.